File: Usuario.java

package parking;


public class Usuario {
    private int id;
    private String nombre;
    private String tipoUsuario; //estudiante y maestro O funcionario

    public Usuario(int id, String nombre, String tipoUsuario) {
        this.id = id;
        this.nombre = nombre;
        this.tipoUsuario = tipoUsuario;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getTipoUsuario() {
        return tipoUsuario;
    }

    public void setTipoUsuario(String tipoUsuario) {
        this.tipoUsuario = tipoUsuario;
    }
    
}

File: ProyectosC#/class/Usuario.cs
public class Usuario
{
    internal object nombre;

    private int Id { get; set; }
    private string Nombre { get; set; }
    private string TipoUsuario { get; set; }   // "Estudiante", "Maestro", "Funcionario"
    private string Carnet { get; set; }

    public Usuario(int id, string nombre, string tipoUsuario, string carnet)
    {
        Id = id;
        Nombre = nombre;
        TipoUsuario = tipoUsuario;
        Carnet = carnet;
    }

    public bool Entrar(Parqueadero parqueadero)
    {
        // Lógica para entrar al parqueadero
        Console.WriteLine($"{Nombre} ha entrado al parqueadero.");
        return true;
    }

    public bool Salir(Parqueadero parqueadero)
    {
        // Lógica para salir del parqueadero
        Console.WriteLine($"{Nombre} ha salido del parqueadero.");
        return true;
    }
}
